src/knx/application_program_object.cpp

```diff
@@ -25,7 +25,7 @@ void ApplicationProgramObject::readProperty(PropertyID id, uint32_t start, uint3
     }
 }
 
-void ApplicationProgramObject::writeProperty(PropertyID id, uint8_t start, uint8_t* data, uint8_t count)
+void ApplicationProgramObject::writeProperty(PropertyID id, uint32_t start, uint8_t* data, uint32_t& count)
```

Comment:
What is the reason for changing the signature? Are the uint8_t parameters not big enough?

Reply:
uint8_t is not enough as startIndex has 12 bits length for A_PropertyValue_* and M_Prop_* services. so uint16_t could be enough for "start".

Concerning "count" I am unsure.
In 3/4/1 Application Interface Layer on page 15 (4.3.1.3 Property Value) it says that every property of a full interface object is always an array. Furthermore, array index 0 is the length of this array. BUT: the size of array index 0 can be either uint16_t or uint32_t depending of the value of "max_no_of_elements" in the property description. If the 4 bit binary exponent is 0, then it is uint16_t, otherwise it is uint32_t.
While writing this, I think that I have mixed up "count" which is basically the argument "nr_of_elements" in the  A_PropertyValue_* and M_Prop_* services and array index 0 which always contains the length of the array: either uint16_t or uint32_t.

Summary for A_PropertyValue_* and M_Prop_* services: 
* "start" (a.k.a. "startIndex in the spec, 12 bits) should be uint16_t
* count (a.k.a. "nr_of_elements, 4 bits) should be uint8_t

However, the interface object server implementation is still broken as it does not handle read/write requests to array index 0 properly as discussed here: https://github.com/thelsing/knx/issues/44

Reply:
Mhmm... after thinking about this again, I think that "count" is just the number of bytes returned in "data". and not "nr_of_elements".

count = nr_of_elements * PropertySize

What is the maximum property size that is defined?

Reply:
See table on page 18. PDT_GENERIC20 has 20 octets:
nr_of_elements (4 bits): 15 * 20octets = 300 octets 
So minimum uint16_t is required.